### PR TITLE
Fix some cppcheck errors

### PR DESCRIPTION
**Pull request by yarrick**
### Contribution description

Fix or suppress some cppcheck errors. Mostly pointer comparison when getting stack sizes or similar.

### Testing procedure

No code change, only comments. The mentioned errors should no longer show in cppcheck run.

### CODE REVIEW

File: cpu/cortexm_common/thread_arch.c [hunk 1]
```diff
@@ -259,10 +259,12 @@ int thread_isr_stack_usage(void)
 {
     uint32_t *ptr = &_sstack;
 
+    /* cppcheck-suppress comparePointers */
     while(((*ptr) == STACK_CANARY_WORD) && (ptr < &_estack)) {
```
Comment (gschorcht) [inline cpu/cortexm_common/thread_arch.c:263]:
```suggestion
    while (((*ptr) == STACK_CANARY_WORD) && (ptr < &_estack)) {
```

Comment (gschorcht) [reply cpu/cortexm_common/thread_arch.c:263]: This should be fixed when we touch the files anyway.

**Review by gschorcht — COMMENTED** [review]
Could you please fix the remaining issues in these files as well, since we are touching these files anyway.

File: cpu/cortexm_common/thread_arch.c (hunk 1 above)
Comment (yarrick) [reply cpu/cortexm_common/thread_arch.c]: Done

**Review by gschorcht — APPROVED** [review]
Looks good

**Pull request merged**